tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go

```diff
@@ -126,78 +116,36 @@ func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioningRKE2Cluster() {
 	}{
-		{"1 Node all roles " + provisioning.AdminClientName.String(), nodeRoles0, r.client, r.psact},
-		{"1 Node all roles " + provisioning.StandardClientName.String(), nodeRoles0, r.standardUserClient, r.psact},
-		{"2 nodes - etcd/cp roles per 1 node " + provisioning.AdminClientName.String(), nodeRoles1, r.client, r.psact},
-		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), nodeRoles1, r.standardUserClient, r.psact},
-		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), nodeRoles2, r.client, r.psact},
-		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), nodeRoles2, r.standardUserClient, r.psact},
+		{"1 Node all roles " + provisioning.AdminClientName.String(), nodeRoles0, r.client},
+		{"1 Node all roles " + provisioning.StandardClientName.String(), nodeRoles0, r.standardUserClient},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.AdminClientName.String(), nodeRoles1, r.client},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), nodeRoles1, r.standardUserClient},
+		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), nodeRoles2, r.client},
+		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), nodeRoles2, r.standardUserClient},
 	}
 
-	var name string
 	for _, tt := range tests {
-		subSession := r.session.NewSession()
-		defer subSession.Cleanup()
-
-		client, err := tt.client.WithSession(subSession)
-		require.NoError(r.T(), err)
-
-		for _, providerName := range r.providers {
-			provider := CreateProvider(providerName)
-			providerName := " Node Provider: " + provider.Name.String()
-			for _, kubeVersion := range r.kubernetesVersions {
-				name = tt.name + providerName + " Kubernetes version: " + kubeVersion
-				for _, cni := range r.cnis {
-					name += " cni: " + cni
-					r.Run(name, func() {
-						TestProvisioningRKE2Cluster(r.T(), client, provider, tt.nodeRoles, kubeVersion, cni, tt.psact, r.advancedOptions)
-					})
-				}
-			}
-		}
+		permutations.RunTestPermutations(&r.Suite, tt.name, tt.client, r.clustersConfig, permutations.RKE2ProvisionCluster, nil, nil)
 	}
 }
 
 func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioningRKE2ClusterDynamicInput() {
-	clustersConfig := new(provisioning.Config)
-	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
-	nodesAndRoles := clustersConfig.NodesAndRoles
-
-	if len(nodesAndRoles) == 0 {
+	if len(r.clustersConfig.NodesAndRoles) == 0 {
 		r.T().Skip()
 	}
 
+	poolNames := permutations.GetTestName(r.clustersConfig.NodesAndRoles)
 	tests := []struct {
 		name   string
 		client *rancher.Client
-		psact  string
 	}{
-		{provisioning.AdminClientName.String(), r.client, r.psact},
-		{provisioning.StandardClientName.String(), r.standardUserClient, r.psact},
+		{provisioning.AdminClientName.String() + poolNames, r.client},
+		{provisioning.StandardClientName.String() + poolNames, r.standardUserClient},
```

Comment:
I still think we shouldn't add the pool names to the test name. I think it would be better to just log it.

Reply:
removed all pool names